### PR TITLE
Add spell check GHA

### LINKED ISSUE #425: Consider adding spell check
I've realized that we don't have a spell check GHA in this repo, but I'm thinking we would benefit from one to check:

- Instruction and README `md` files
- QC report `Rmd` files

This issue is brought to you by me realizing I'm writing `celltype` and `cell type` interchangeably (not good!) while preparing the cell type QC report, and realizing that a spell checker would therefore be useful.

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #425 

This PR adds spell check, and fixes typos along the way! I set up the action to run on PRs to `main` and `development`. So, to check the action, let's wait to approve/merge this until its base is `development` and the workflow can run through. I'm therefore opening this as a draft PR, and will remove draft status after #422, which this is currently stacked on, is merged.

One thing I didn't decide on is whether we might have a preference for "fastq" vs "FASTQ". It didn't seem to matter too much either, so I kept them both. Is there a preference either way?

I recommend clicking "hide whitespace" during review, since VSCode ate up lots of spaces here..

### CODE REVIEW

**Comment by sjspielman** [comment]
Spell check workflow is all tested now :tada:

File: components/dictionary.txt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+glioblastoma
+GRCh
+Hippen
+HTOs
+HPC
+intronic
+Lun
+miQC
+nextflow
+Nextflow
```
Comment (allyhawkins) [inline components/dictionary.txt:31]: I thought the spell checker didn't care about capitalization? Or am I remembering incorrectly?

Comment (sjspielman) [reply components/dictionary.txt:31]: I never remember correctly about this 😭.. will check!

Comment (sjspielman) [reply components/dictionary.txt]: What I learned (re-learned?) is that lowercase will match the uppercase, but not vice versa. So, I deleted the uppercase in https://github.com/AlexsLemonade/scpca-nf/pull/426/commits/15cb26457a1ad96007cd3005fe09d968b8f98524

**Review by allyhawkins — APPROVED** [review]
looks good, thanks for adding this!

**Pull request merged**